source/solvers/gd_navier_stokes.cc

```diff
@@ -110,12 +110,24 @@ GDNavierStokesSolver<dim>::setup_assemblers()
             this->simulation_parameters.physical_properties));
         }
 
-      // Core assembler
-      this->assemblers.push_back(
-        std::make_shared<GDNavierStokesAssemblerCore<dim>>(
-          this->simulation_control,
-          this->simulation_parameters.physical_properties,
-          gamma));
+      if (this->simulation_parameters.physical_properties.non_newtonian_flow)
+        {
+          // Core assembler with Non newtonian viscosity
+          this->assemblers.push_back(
```

Comment:
I love this notion of having a single assembler for all non-newtonian flow, this is very nice.

Reply:
Thank you.